sys/random/fortuna/fortuna.h

```diff
@@ -0,0 +1,226 @@
+ *        delay reseeds even if good random entropy is collected. In section
+ *        9.5.5, a suitable value of 64 bytes is suggested.
+ */
+#ifndef FORTUNA_MIN_POOL_SIZE
+#define FORTUNA_MIN_POOL_SIZE       (64U)
+#endif
+
+
+/**
+ * @brief Definition of the seed file size used to initialize the PRNG. The
+ *        default value of 64 bytes is suggested by section 9.6.1.
+ */
+#ifndef FORTUNA_SEED_SIZE
+#define FORTUNA_SEED_SIZE           (64U)
+#endif
+
+/**
+ * @brief Reseed interval in us. After this interval, the PRNG must be
+ *        reseeded. Per section 9.5.5, the recommended value is 100ms. Set to
+ *        zero to disable this security feature.
+ */
+#ifndef FORTUNA_RESEED_INTERVAL
+#define FORTUNA_RESEED_INTERVAL     (0)
+#endif
+
+/**
+ * @brief Reseed limit in bytes. After this number of bytes, the PRNG must be
+ *        reseeded. Per section 9.4.4, the recommended value is 2^20 bytes
+ *        (=1 MB). Set to zero to disable this security feature.
+ */
+#ifndef FORTUNA_RESEED_LIMIT
+#define FORTUNA_RESEED_LIMIT        (1U << 20)
+#endif
+
+/**
+ * @brief Fortuna requires a block cipher with a 256-bit key to minimize
+ *        permutation attacks.
+ *
+ * @note  RIOT-OS does not (yet) support AES-256. Therefore, at the expense of
+ *        security, AES-128 is used instead. An detailed explanation can be
+*         found at https://crypto.stackexchange.com/a/5736.
+ */
+#ifndef FORTUNA_AES_KEY_SIZE
+#define FORTUNA_AES_KEY_SIZE        (16U)
+#endif
+
+/**
+ * @brief Use a mutex to lock concurrent access when running sensitive
+ *        operations in multi-threaded applications.
+ */
+#ifndef FORTUNA_LOCK
+#define FORTUNA_LOCK                (1)
+#endif
+
+/**
+ * @brief When set, add additional instructions to clear temporary variables
+ *        to prevent leaking sensitive information.
+ */
+#ifndef FORTUNA_CLEANUP
+#define FORTUNA_CLEANUP             (1)
+#endif
+
+/**
+ * @brief Generator counter and key.
+ */
+typedef struct {
+    uint8_t key[32];
+    union {
+        struct {
+            uint64_t l;
+            uint64_t h;
+        } split;
+        uint8_t bytes[16];
+    } counter;
+} fortuna_generator_t;
+
+/**
+ * @brief Pool for storing entropy.
+ */
+typedef struct {
+    sha256_context_t ctx;
+    uint32_t len;
+} fortuna_pool_t;
+
+/**
+ * @brief The Fortuna state.
+ */
+typedef struct {
+    fortuna_generator_t gen;
+    fortuna_pool_t pools[FORTUNA_POOLS];
+    uint32_t reseeds;
+#if FORTUNA_RESEED_INTERVAL > 0
+    uint64_t last_reseed;
+#endif
+#if FORTUNA_LOCK
+    mutex_t lock;
+#endif
+} fortuna_state_t;
+
+/**
+ * @brief Type definition of a Fortuna seed file.
+ */
+typedef uint32_t fortuna_seed_t[FORTUNA_SEED_SIZE];
+
+/**
+ * @brief   Initialize the Fortuna PRNG state.
+ *
+ * It is possible to use this method to clear an existing state.
+ *
+ * @param[in] state         PRNG state
+ * @return                  zero on succesful initialization
```

Comment:
here and below, add empty line between param and return

Reply:
Done.